### DISCUSSION TITLE
howto: reference nested document from other collection

### DISCUSSION THREAD

**Discussion by RatakondalaArun** [typegoose/typegoose#663] [Q&A] [answered]
Hi

I want to auto-populate a field with a reference to nested document from other collection

- `User` and `Class` are top-level collections
- `Subject` is a nested document of `Class` collection

```ts
// user.ts
// users collection
class User{
  _id: ObjectId;
  @prop()
  name: string;

  @prop({
    ref: () => Class,
    localField: 'subject',
    foreignField: 'subjects._id',
    autopopulate: { maxDepth: 1 },
    justOne: true,
  })
  subject: Ref<Subject,ObjectId>; // <-- this is the field that I want to populate from class.subjects
}
```
---
```ts
// class.ts
// classes collection
class Class{
  _id: ObjectId;
  @prop({type:()=>[Subject]})
  subjects: Subject[];
}
```
---
```ts
// subject.ts
// nested document from `classes` collection
class Subject{
 _id:ObjectId;
  @prop()
  name: string;
}
```

**Comment by hasezoey** [comment, marked as answer]
>     localField: 'subject',

from what i know, you cant use the same field to store a reference and at the same time use as a virtual populate

---

also, from what i know, yes you can *match* against nested documents, but i dont know of how to / or if *ignoring the parent* is possible

if you really just want `Subject`, maybe consider making it into its own collection and reference it from the other documents that require it
